### PR TITLE
WIP: BXMSDOC-6414 Immutable Spring Boot Deployment

### PR DESCRIPTION
**Pull request by emmurphy1**

### CODE REVIEW

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+.Procedure
+. Add the following properties to the `service/source/main/resources/application.properties` file of your Spring Boot application:
++
+[source]
+----
+kieserver.classPathContainer=true
+kieserver.autoScanDeployments=true
+----
++
+When both of these properties are set to `true`, Maven automatically scans for container IDs that contain `<artifactId>` and `<version>` at build time.
```
Comment (elguardian) [inline doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc]: not maven, spring boot kie server
and it scans for deployable artifacts that were packaged within spring boot application using
package-dependencies-kjar maven goal of kie-maven-plugin

**Review by afalhambra — COMMENTED** [review]
Nice work @emmurphy1 - left some comments. Please, have a look and let me know

**Review by michelehaglund — COMMENTED** [review]
" - i.e. same Maven coordinates except version -" should it be for example?

Not letting me comment in line.

**Comment by afalhambra** [comment]
> " - i.e. same Maven coordinates except version -" should it be for example?
> 
> Not letting me comment in line.

@michelehaglund it was just a small clarification about what we mean by "multiple versions of the same KJAR". But maybe it's clear enough and we don't need that clarification. It's up to you.

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,98 @@
+[id='creating-self-contained-image-proc_{context}']
+= Creating a self-contained {PRODUCT} Spring Boot JAR file
```
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc:2]: @afalhambra, Enrique suggested that 'immutable' has an unwanted (in this case) associated with Open Shift. WYDT about 'unchangeable':
Creating a self-contained and unchangeable ....

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc]: I know what you mean, @afalhambra. I think you have to look at the Files tab. That shows the latest changes.

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+.Prerequisites
+* You have a {PRODUCT} Spring Boot project.
+
+.Procedure
+. Add the following properties to the `service/source/main/resources/application.properties` file of your Spring Boot application:
++
+[source]
+----
+kieserver.classPathContainer=true
+kieserver.autoScanDeployments=true
```
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc:15]: Yes, we generally only show one way unless there is a strong argument to show both. Please give me a descriptions of what these properties are for and I'll add it.

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (afalhambra) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc:2]: Ok, I see, no worries then. No changes needed here. Thanks @emmurphy1

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,98 @@
+[id='creating-self-contained-image-proc_{context}']
+= Creating a self-contained {PRODUCT} Spring Boot JAR file
+
+You can create a self-contained {PRODUCT} Spring Boot JAR file that contains multiple versions of the same KJAR file, including modules. The KJAR files are separated from the `BOOT-INF/lib` JAR files to avoid class loader collisions. Each classpath container KJAR file is isolated for other KJAR files and does not rely on the Spring Boot class loader.
+
+.Prerequisites
+* You have a {PRODUCT} Spring Boot project.
+
+.Procedure
+. Add the following properties to the `service/source/main/resources/application.properties` file of your Spring Boot application:
```
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc]: I still like 'properties' better than lines.

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (afalhambra) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc:15]: Ok, thanks for clarifying.
 
kieserver.classPathContainer: When set to `true` kie server will use the class loader used by the container in order to load KJAR files and its dependencies.
kieserver.autoScanDeployments: When set to `true` kie server will scan and deploy all KJAR files available in the Spring Boot application, either declared programmatically or through the Maven plugin

@elguardian please feel free to modify or correct my above descriptions of these properties.

File: doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,98 @@
+[id='creating-self-contained-image-proc_{context}']
+= Creating a self-contained {PRODUCT} Spring Boot JAR file
+
+You can create a self-contained {PRODUCT} Spring Boot JAR file that contains multiple versions of the same KJAR file, including modules. The KJAR files are separated from the `BOOT-INF/lib` JAR files to avoid class loader collisions. Each classpath container KJAR file is isolated for other KJAR files and does not rely on the Spring Boot class loader.
```
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc]: @elguardian  would clarify what 'modules' means, above? Or can we delete 'including modules'?

**Pull request merged**